### PR TITLE
Added the matplotlib 1.5 release candidate.

### PR DESCRIPTION
**Pull request by pelson**
Most of this has been taken directly from https://github.com/conda/conda-recipes/tree/master/matplotlib.
At some point, it makes sense to make optional matplotlib packages which add the appropriate dependencies for specific backends (e.g. ```conda install matplotlib[-nbagg]``` or ```conda install matplotlib[-qtagg]```)

### CODE REVIEW

**Comment by pelson** [comment]
It would also be nice for this to target a dev channel on conda-forge, or something...

**Comment by scopatz** [comment]
hmm seems like it is failing :(

**Comment by pelson** [comment]
> hmm seems like it is failing :(

Just a bit :wink:

**Comment by pelson** [comment]
@tacaswell - I failed miserably trying to build this stuff on basic installations. On the other hand, I think it would be pretty sweet if we had builds for mpl on all platforms within a few minutes of tagging the release...

**Comment by tacaswell** [comment]
I got mpl to build on linux with the recipe here: https://github.com/conda/conda-recipes/pull/419

**Comment by pelson** [comment]
That was the basis of this PR. If we can get this working, we have CI for Linux, OSX and Windows, and we would be shipping to the conda-forge channel... :wink:

**Comment by pelson** [comment]
The Windows build is failing with: ```src/ft2font.cpp(5) : fatal error C1083: Cannot open include file: 'string': No such file or directory```, after the following header information:

```
The following NEW packages will be INSTALLED: 
 
    certifi:            14.05.14-py27_0 defaults                           
    cycler:             0.9.0-py27_0    file:///C|/conda/conda-bld/win-64/ 
    freetype:           2.4.10-0        defaults                           
    libpng:             1.6.17-0        defaults                           
    nose:               1.3.7-py27_0    defaults                           
    numpy:              1.9.2-py27_0    defaults                           
    pip:                7.1.2-py27_0    defaults                           
    pyparsing:          2.0.3-py27_0    defaults                           
    pyqt:               4.10.4-py27_1   defaults                           
    python:             2.7.10-0        defaults                           
    python-dateutil:    2.4.2-py27_0    defaults                           
    pytz:               2015.4-py27_0   defaults                           
    setuptools:         18.1-py27_0     defaults                           
    six:                1.9.0-py27_0    defaults                           
    ssl_match_hostname: 3.4.0.2-py27_0  defaults                           
    tornado:            4.2.1-py27_0    defaults                           
    wheel:              0.24.0-py27_0   defaults                           
    zlib:               1.2.8-0         defaults   

BUILDING MATPLOTLIB 
            matplotlib: yes [1.5.0rc1] 
                python: yes [2.7.10 |Continuum Analytics, Inc.| (default,
                        May 28 2015, 16:44:52) [MSC v.1500 64 bit (AMD64)]] 
              platform: yes [win32] 
 
REQUIRED DEPENDENCIES AND EXTENSIONS 
                 numpy: yes [version 1.9.2] 
              dateutil: yes [using dateutil version 2.4.2] 
                  pytz: yes [using pytz version 2015.4] 
                cycler: yes [using cycler version 0.9.0] 
               tornado: yes [using tornado version 4.2.1] 
             pyparsing: yes [using pyparsing version 2.0.3] 
                libagg: yes [pkg-config information for 'libagg' could not
                        be found. Using local copy.] 
              freetype: yes [Using unknown version found on system.] 
                   png: yes [Using unknown version found on system.] 
                 qhull: yes [pkg-config information for 'qhull' could not be
                        found. Using local copy.] 
 
OPTIONAL SUBPACKAGES 
           sample_data: yes [installing] 
              toolkits: yes [installing] 
                 tests: yes [using nose version 1.3.7 / mock is required to
                        run the matplotlib test suite. "setup.py test" will
                        automatically download it. Install mock to run
                        matplotlib.test()] 
        toolkits_tests: yes [using nose version 1.3.7 / mock is required to
                        run the matplotlib test suite. "setup.py test" will
                        automatically download it. Install mock to run
                        matplotlib.test()] 
 
OPTIONAL BACKEND EXTENSIONS 
                macosx: no  [Mac OS-X only] 
                qt5agg: no  [PyQt5 not found]
                qt4agg: no  [Check timed out]
               gtk3agg: no  [Requires pygobject to be installed.]
             gtk3cairo: no  [Requires cairocffi or pycairo to be installed.] 
                gtkagg: no  [Requires pygtk]
                 tkagg: no  [The C/C++ header for Tk (tk.h) could not be
                        found.  You may need to install the development
                        package.] 
                 wxagg: no  [requires wxPython]
                   gtk: no  [Requires pygtk]
                   agg: yes [installing]
                 cairo: no  [cairocffi or pycairo not found]
             windowing: yes [installing, installing]
 
OPTIONAL LATEX DEPENDENCIES
                dvipng: no 
           ghostscript: no 
                 latex: no 
               pdftops: no
```

@cgohlke, did you see any compilation issues with v1.5.0rc1? I've tried to avoid installing any system libraries, but do have anaconda's freetype. I wouldn't rule out that being packaged strangely though.

**Comment by cgohlke** [comment]
No problems here. For Python 2.7 I am using "Visual Studio 2008 Pro" (not "Visual C++ for Python") and [stdint.h from msinttypes](https://msinttypes.googlecode.com/svn/trunk/stdint.h).

**Comment by pelson** [comment]
> No problems here. For Python 2.7 I am using "Visual Studio 2008 Pro" (not "Visual C++ for Python") and stdint.h from msinttypes.

Thanks for the pointer @cgohlke. I ended up packaging msinttypes which fixed my build issue.

**Comment by pelson** [comment]
Success to some degree. I have builds on Win32, Linux64 and OSX. I'm going to bank these, and iterate in a subsequent PR.

Keep an eye out for anaconda.org/conda-forge - there should be some mpl builds turning up soon!

**Comment by scopatz** [comment]
woohoo!

**Comment by pelson** [comment]
> woohoo!

Definitely. :tada: 
@tacaswell - hope this can help us cycle through the RCs with a little less pain.

**Comment by ocefpaf** [comment]
> I ended up packaging msinttypes which fixed my build issue.

This is awesome.  We have a few packages that needs that and solution was to copy `stdint.h`.  I will start using that package instead.

File: recipes/matplotlib/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    - setuptools
+    - numpy *
+    - python-dateutil
+    - freetype
+    - msinttypes          # [win]
+    - cycler
+    - nose
+    - pyparsing
+    - pytz
+#    - py2cairo            # [linux and py2k]
```
Comment (JanSchulz) [inline recipes/matplotlib/meta.yaml]: I'm currently running into a similar problem: could it be that obvious-ci does not set CONDA_PY when calling the conda build procedure? My observation was that the wrong py versions are set so this selector is only right when the "outer" conda version is the same as the one which should build the thing...

It works on windows, as there `CONDA_PY` is set via the build matrix.

Comment (pelson) [inline recipes/matplotlib/meta.yaml]: I don't think so, that will be set as python is explicitly a build & run dependency.

**Pull request merged**